features/FEATURE_LWIP/lwip-interface/lwip-sys/lwip_random.c

```diff
@@ -0,0 +1,50 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "lwip/opt.h"
+#include "lwip/debug.h"
+#include "lwip/def.h"
+
+#include "lwip_random.h"
+
+#if FEATURE_COMMON_PAL
+
+#include "ns_types.h"
+#include "randLIB.h"
+
+void lwip_seed_random(void)
+{
+    randLIB_seed_random();
+}
+
+void lwip_add_ramdom_seed(uint64_t seed)
+{
+    randLIB_add_seed(seed);
+}
+
+inline uint32_t lwip_get_random(void)
```

Comment:
Only having inline in the C file doesn't achieve anything.

If you really want inlining, the only easily C99/C++ portable form is a "static inline" definition in the header file.